src/Controller/UserController.php

```diff
@@ -29,33 +38,54 @@ public function index(UserRepository $userRepository): Response
     }
 
     /**
+     * @param Request $request
+     * @param UserPasswordEncoderInterface $passwordEncoder
      * @Route("/new", name="new", methods={"GET","POST"})
+     * @return Response
+     * @throws \Twig_Error_Loader
+     * @throws \Twig_Error_Runtime
+     * @throws \Twig_Error_Syntax
      */
     public function new(Request $request, UserPasswordEncoderInterface $passwordEncoder): Response
     {
         $user = new User();
         $form = $this->createForm(UserType::class, $user, ['group' => 'new']);
         $form->handleRequest($request);
-
         if ($form->isSubmitted() && $form->isValid()) {
 
+            //generate and set email_token (for the email checking)
+            $token = $this->emailService->gen_uuid();
+            $user->setEmailToken($token);
+
+            //sending the confirmation email
+            if(!$this->emailService->confirmRegistration($user->getLogin(), $user->getEmail(), $user->getEmailToken())) {
+                return $this->render('user/new.html.twig', [
+                    'user' => $user,
+                    'form' => $form->createView()
+                ]);
+            }
+
             $user->setPassword(
                 $passwordEncoder->encodePassword(
                     $user,
                     $form->get('password')->getData()
                 )
             );
 
+
+
             $entityManager = $this->getDoctrine()->getManager();
             $entityManager->persist($user);
             $entityManager->flush();
 
+
+
             return $this->redirectToRoute('app_user_index');
         }
 
         return $this->render('user/new.html.twig', [
             'user' => $user,
-            'form' => $form->createView(),
+            'form' => $form->createView()
```

Comment:
In general it's better to leave the last comma (usually called trailing comma):
right now if I add a new key to the array, I have to modify two lines: the currently last one to add the comma, and the next one to add my key/value. If you leave the trailing comma, I'll only have to add a line ! It makes PRs a bit easier to read !

Reply:
Done, thanks for the tips :)